### ISSUE TITLE
d3d12core should be added to VKD3DManager.managed_dlls

### ISSUE THREAD

**Issue by mihailim** [lutris/lutris#4782]
A couple of weeks ago, in [PR1458](https://github.com/HansKristian-Work/vkd3d-proton/pull/1458), vkd3d-proton moved the D3D12 implementation out of `d3d12.dll` and into a companion `d3d12core.dll`, turning `d3d12.dll` into a thin wrapper around the other one. Since Lutris' VKD3DManager doesn't yet know about it, it won't copy the actual implementing DLL, just the wrapper, causing runtime link failures for D3D12 consumers.

The fix is trivial, adding "d3d12core" to `managed_dlls` in `lutris/util/wine/vkd3d.py` is sufficient. This can be done safely without impacting existing / older vkd3d builds, since `DLLManager` will invoke `disable_dll` for non-existent DLLs.

In essence, the change is just

```patch
--- vkd3d.py.orig       2022-12-01 23:32:28.000000000 +0200
+++ vkd3d.py    2023-03-24 09:48:14.691788892 +0200
@@ -8,5 +8,5 @@
     component = "VKD3D"
     base_dir = os.path.join(RUNTIME_DIR, "vkd3d")
     versions_path = os.path.join(base_dir, "vkd3d_versions.json")
-    managed_dlls = ("d3d12", )
+    managed_dlls = ("d3d12", "d3d12core", )
     releases_url = "https://api.github.com/repos/lutris/vkd3d/releases"
```

Thanks!

**Comment by sim590** [comment]
It seems like Lutris updates its `vkd3d-proton` to the latest for every game by default or am I mistaken? And this change does just break games with no easy to understand explanation. The following appears when trying to launch Diablo 2 Resurrected for instance:

```
4203.479:0794:0798:err:vkd3d-proton:load_d3d12core_once: Failed to find vkd3d-proton d3d12core interfaces. Make sure d3d12core.dll is installed as well.
```

Maybe submodules that introduce this type of changes should advertise the requirement of a certain version of Lutris to update. This way, Lutris could avoid updating versions that could break games.

Right now, on Archlinux Lutris is at version 0.5.12 (package is flagged out of date), but vkd3d-proton updates itself through Lutris and games break.

## Note to any user hitting this

either update to Lutris 0.5.13 or downgrade vkd3d-proton to 2.8 (in the game config) in order to avoid this.

**Comment by danieljohnson2** [comment]
Yep, Lutris totally does this. You can downgrade VKD3D in the Runner options to affect all your games at once, too.

The official Lutris policy is that we support the latest release only; that means the website will make changes that aren't compatible with older Lutris versions, and the solution for that is that users should upgrade to the latest Lutris.

Lutris could do more to cater to older clients- it could tag components with the earliest compatible Lutris version, and future clients could care about this. But this requires both client and server to be updated, and it's a policy change that's above my pay grade for sure.

Instead, we might have the server provide the .dll lists use by the components, so that can be updated server side. That still requires both ends to be upgraded to get it working, but it isn't a policy change at least. It would have allowed the latest vkd3d to go out before 0.5.13 did. This would address one narrow compatibility issue though, and not a common one from what I've seen.

But even if we do this, it's not all roses and song. Upgrading to anew VKD3D (or whatnot) can break your games even if Lutris itself is compatible. Maybe Lutris instead should not allow components used in games to be upgraded implicitly, but I'm not too sure *that* would be welcome at all.

It is a puzzlement.

**Comment by mihailim** [comment]
That's certainly one option. Another one would be to provide a DLL manifest (e.g. JSON or whatnot) inside the tarball packaging the component. That way, there wouldn't be any reliance on remote availability of that manifest. This would be backwards compatible with older clients, since they'd just ignore the extra file.

**Comment by danieljohnson2** [comment]
Hmm. That's not a bad idea at all. The server would not need to care, I think- we'd install the most recent version, realize it's not compatible after we see what's in the archive, and then back off to an older version.

One down side is that a very old Lutris could download many, many versions of a component before it finds a compatible one. I wonder if I can short circuit that.
